vaadin-map-flow-parent/vaadin-map-flow/src/main/java/com/vaadin/flow/component/map/Map.java

```diff
@@ -111,16 +112,7 @@ public class Map extends MapBase {
      *            the user projection to use for all public facing API
      */
     public static void setUserProjection(String projection) {
-        UI ui = UI.getCurrent();
-        if (ui == null || ui.getPage() == null) {
-            throw new IllegalStateException("UI instance is not available. "
-                    + "It means that you are calling this method "
-                    + "out of a normal workflow where it's always implicitly set. "
-                    + "That may happen if you call the method from the custom thread without "
-                    + "'UI::access' or from tests without proper initialization.");
-        }
-        UI.getCurrent().getPage().executeJs(
-                "window.Vaadin.Flow.mapConnector.setUserProjection($0)",
+        executeJs("window.Vaadin.Flow.mapConnector.setUserProjection($0)",
```

Comment:
IMO, the code looks clearer without extracting this into a helper:

```suggestion
        UI.getCurrentOrThrow().getPage().executeJs("window.Vaadin.Flow.mapConnector.setUserProjection($0)",
```

Reply:
Updated